code/modules/projectiles/guns/ballistic.dm

```diff
@@ -70,16 +70,18 @@
 			to_chat(user, "<span class='notice'>There's already a magazine in \the [src].</span>")
 	if(istype(A, /obj/item/suppressor))
 		var/obj/item/suppressor/S = A
-		if(can_suppress)
+		if(!user.is_holding(src))
+			to_chat(user, "<span class='notice'>You need be holding [src] to fit [S] to it!</span>")
+			return
+		else if(can_suppress)
 			if(!suppressed)
 				if(!user.transferItemToLoc(A, src))
 					return
 				to_chat(user, "<span class='notice'>You screw [S] onto [src].</span>")
 				suppressed = A
 				S.oldsound = fire_sound
-				S.initial_w_class = w_class
 				fire_sound = 'sound/weapons/gunshot_silenced.ogg'
-				w_class = WEIGHT_CLASS_NORMAL //so pistols do not fit in pockets when suppressed
+				w_class += A.w_class //so pistols do not fit in pockets when suppressed
```

Comment:
I feel that guns that are already w_class 3 or greater should stay the way they are when suppressed.

Reply:
My rationale behind this is that it means that attaching a suppressor to a sniper rifle or a C20r for example would make it no longer fit in your backpack due to the added mass, and removing the suppressor makes it fit again.

This adds a new game play element where you have to decide wither to leave that suppressor on and have to hold your weapon, or end up having to show off your large weapon to the world with the trade-off of it not showing messages & being quieter.

Reply:
You could just simply unsuppress your gun to put it in your backpack, and shove the suppressor in the backpack alongside it. Then tada, ya cheesed that system, as you can simply take the gun and suppressor out, then slap the suppressor right back on the gun for tacticool silenced memery.

Reply:
That's not cheesing the system, that IS the system.